### PR TITLE
Add string attribute api to text span

### LINKED ISSUE #79318: No available api to make talkback or voiceover to spellout characters
b/182520487
b/182336500
b/182338851

## Steps to Reproduce

```dart
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    const title = 'code sample';
    return MaterialApp(
      title: title,
      home: Scaffold(
        body: Home(),
      ),
    );
  }
}

class Home extends StatelessWidget {
  const Home({Key key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Center(
      child: Column(
        mainAxisAlignment: MainAxisAlignment.center,
        children: [
          Text('IwantThisSpellOut'),
          Text('IwantThisSpellOut'),
        ],
      ),
    );
  }
}
```

There is no API in flutter to make the talkback or voiceover to announce character by character if the text receives focus.

Design doc: [flutter.dev/go/a11y-text-attributes](flutter.dev/go/a11y-text-attributes)

### PR DESCRIPTION
**Pull request by chunhtai**
fixes https://github.com/flutter/flutter/issues/79318

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by chunhtai** [comment]
cc @Renzo-Olivares @justinmc  in case this may overlap with the attributed text for rich text

File: packages/flutter/lib/src/painting/text_span.dart [hunk 1]
```diff
@@ -218,6 +220,27 @@ class TextSpan extends InlineSpan implements HitTestTarget, MouseTrackerAnnotati
...
+  /// Whether the assistive technologies should spell out this text character
+  /// by character
+  ///
+  /// If the text is 'hello world', setting this to true causes the assistive
+  /// technologies, such as VoiceOver or TalkBack, to pronounce
+  /// 'h-e-l-l-o-space-w-o-r-l-d' instead of complete words. This is useful for
+  /// texts, such as passwords or verification codes.
+  ///
+  /// If this span contains other text span children, they are also inherited
+  /// the property from this span unless explicitly set.
```
Comment (justinmc) [inline packages/flutter/lib/src/painting/text_span.dart:241]: Nit: Up to you, but maybe consider adding a sentence about the default behavior if this isn't specified.

**Review by justinmc — APPROVED** [review]
LGTM 👍 

Nothing jumps out at me as affecting rich text.  I left one comment about a potential performance concern if we do have large documents, though.

**Comment by chunhtai** [comment]
Added caching logic plus some more test for selectable text, @justinmc  PTAL

File: packages/flutter/lib/src/rendering/editable.dart [hunk 2]
```diff
@@ -2737,10 +2741,31 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin,
...
+            attributes.add(
+              infoAttribute.copy(
+                range: TextRange(start: offset + originalRange.start, end: offset + originalRange.end),
+              ),
+            );
+          }
+          buffer.write(label);
+          offset += label.length;
+        }
+        _cachedAttributedValue = AttributedString(buffer.toString(), attributes: attributes);
```
Comment (justinmc) [inline packages/flutter/lib/src/rendering/editable.dart:2764]: Nice, I'm glad this could be cached 👍 .  And in RenderParagraph.

**Review by justinmc — APPROVED** [review]
LGTM with the new changes, thanks! 👍

**Pull request merged**